spotbugs/src/main/java/edu/umd/cs/findbugs/BugPattern.java

```diff
@@ -93,13 +97,20 @@ public BugPattern(String type, String abbrev, String category, boolean experimen
         this.url = bugsUrl;
     }
 
-    static final BugPattern REALLY_UNKNOWN = new BugPattern("REALLY_UNKNOWN", "TEST", "CORRECTNESS", false,
+    static final BugPattern REALLY_UNKNOWN = new BugPattern("SPOT0000","REALLY_UNKNOWN", "TEST", "CORRECTNESS", false,
             "Unknown warning; core bug patterns not found", "Unknown warning BUG_PATTERN in {1}; core bug patterns not found",
             "<p>A warning was recorded, but SpotBugs can't find the description of this bug pattern "
                     + "and so can't describe it. This should occur only in cases of a bug in SpotBugs or its configuration, "
                     + "or perhaps if an analysis was generated using a plugin, but that plugin is not currently loaded. "
                     + "</p>", null, 0);
 
+    /**
+     * Get the Rule Id
+     */
+    public String getRuleId() {
```

Comment:
If we allow plugins to not providing the `ruleId` attribute, annotate this method with [`@CheckForNull`](https://spotbugs.readthedocs.io/en/stable/annotations.html#edu-umd-cs-findbugs-annotations-checkfornull).